liota/dcc_comms/mqtt_dcc_comms.py

```diff
@@ -124,17 +125,32 @@ def _disconnect(self):
         """
         self.client.disconnect()
 
-    def subscribe(self, msg_attr=None):
+    def receive(self, msg_attr=None):
         """
         Subscribes to a topic with specified QoS and callback.
+        Set call back to receive_message method if no callback method is passed by user.
 
         :param msg_attr: MqttMessagingAttributes Object
         :return:
         """
+        if msg_attr is not None and hasattr(msg_attr.sub_callback):
```

Comment:
`hasattr` check is redundant as `MqttMessagingAttributes` object will have it.

This way is more pythonic :)

```
callback = msg_attr.sub_callback if msg_attr else self.receive_message
```